### PR TITLE
use Services.telemetry.canRecordBase, enable snippets by default

### LINKED ISSUE #3007: Snippets: Read telemetryEnabled from Services.telemetry.canRecordBase.
Also see https://bugzilla.mozilla.org/show_bug.cgi?id=1384843
Based on suggestion given by  :gfritzsche in [Bug 1384843](https://bugzilla.mozilla.org/show_bug.cgi?id=1384843) we should get `ervices.telemetry.canRecordBase` instead of directly reading prefs.

/cc @k88hudson

### LINKED ISSUE #3056: Enable Snippets by default
Now that we have some templates, let's turn on snippets by default.

### PR DESCRIPTION
**Pull request by k88hudson**
Fix #3056 and fix #3007.

I've updated the PR to also fix TelemetrySender and related tests, as well as updating the value on every new tab init.'

I also snuck a `.hasFxAccount` property into snippets data

### CODE REVIEW

File: system-addon/lib/SnippetsFeed.jsm [hunk 1]
```diff
@@ -47,16 +47,17 @@ this.SnippetsFeed = class SnippetsFeed {
...
       telemetryEnabled: Services.telemetry.canRecordBase,
-
       onboardingFinished: Services.prefs.getBoolPref(ONBOARDING_FINISHED_PREF)
     };
 
     this.store.dispatch(ac.BroadcastToContent({type: at.SNIPPETS_DATA, data}));
   }
+  _refreshCanRecordBase() {
+    // TODO: There is currently no way to listen for changes to this value;
+    // bug is filed here https://bugzilla.mozilla.org/show_bug.cgi?id=1386318
```
Comment (dmose) [inline system-addon/lib/SnippetsFeed.jsm]: Maybe mention that for now, we're doing the best effort of calling this each time a new tab is initialized.

File: system-addon/lib/SnippetsFeed.jsm [hunk 2]
```diff
@@ -15,6 +15,7 @@ XPCOMUtils.defineLazyModuleGetter(this, "ProfileAge",
 // Url to fetch snippets, in the urlFormatter service format.
 const SNIPPETS_URL_PREF = "browser.aboutHomeSnippets.updateUrl";
 const ONBOARDING_FINISHED_PREF = "browser.onboarding.notification.finished";
+const FXA_USERNAME_PREF = "services.sync.username";
```
Comment (dmose) [inline system-addon/lib/SnippetsFeed.jsm]: Does sending this need data review?

Comment (k88hudson) [inline system-addon/lib/SnippetsFeed.jsm]: Because it's already implemented in existing snippets, I would assume not

**Comment by Mardak** [comment]
Do we want to hold off on merging this? Or have a temporary mozilla-central-patches to prevent pine failures see https://bugzilla.mozilla.org/show_bug.cgi?id=1386350 ?

**Comment by Mardak** [comment]
Eh, actually last time we had a separate PR to add the patch. That make it simple to revert that one PR.

**Comment by as-pine-proxy** [comment]
[Treeherder: 7c30c1f808005445be8ab4b87ec1293bcf3abe98](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=7c30c1f808005445be8ab4b87ec1293bcf3abe98)

**Pull request merged**